java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/launchers/JUnitLaunchConfigurationDelegate.java

```diff
@@ -167,30 +108,97 @@ private ITestKind getTestRunnerKind(ILaunchConfiguration configuration) {
+
+        return arguments.toArray(new String[arguments.size()]);
+    }
+
+    private void addTestItemArgs(List<String> arguments) throws CoreException {
+        if (this.args.testLevel == TestLevel.CLASS) {
+            final String fileName = createTestNamesFile(this.args.testNames);
+            arguments.add("-testNameFile");
+            arguments.add(fileName);
+        } else if (this.args.testLevel == TestLevel.METHOD) {
+            arguments.add("-test");
+            final IMethod method = (IMethod) JavaCore.create(this.args.testNames[0]);
+            String testName = method.getElementName();
+            if (this.args.testKind == TestKind.JUnit5 && method.getParameters().length > 0) {
+                final ICompilationUnit unit = method.getCompilationUnit();
+                if (unit == null) {
+                    throw new CoreException(new Status(IStatus.ERROR, JUnitPlugin.PLUGIN_ID, IStatus.ERROR,
+                            "Cannot get compilation unit of method" + method.getElementName(), null)); //$NON-NLS-1$
+                }
+                final CompilationUnit root = (CompilationUnit) TestSearchUtils.parseToAst(unit,
+                        false /*fromCache*/, new NullProgressMonitor());
+                final String key = method.getKey();
+                ASTNode methodDeclaration = root.findDeclaringNode(key);
+                if (methodDeclaration == null) {
+                    // fallback to find it according to source range
+                    methodDeclaration = NodeFinder.perform(root, method.getSourceRange().getOffset(),
+                            method.getSourceRange().getLength(), unit);
+                }
+                if (!(methodDeclaration instanceof MethodDeclaration)) {
+                    throw new CoreException(new Status(IStatus.ERROR, JUnitPlugin.PLUGIN_ID, IStatus.ERROR,
+                            "Cannot get method declaration of method" + method.getElementName(), null)); //$NON-NLS-1$
+                }
+
+                final List<String> parameters = new LinkedList<>();
+                for (final Object obj : ((MethodDeclaration) methodDeclaration).parameters()) {
+                    if (obj instanceof SingleVariableDeclaration) {
+                        final ITypeBinding paramTypeBinding = ((SingleVariableDeclaration) obj)
+                                .getType().resolveBinding();
+                        if (paramTypeBinding.isParameterizedType()) {
+                            parameters.add(paramTypeBinding.getBinaryName());
+                        } else {
+                            parameters.add(paramTypeBinding.getQualifiedName());
+                        }
                     }
-                } catch (IOException | CoreException e) {
-                    // do nothing
                 }
-                return;
+                if (parameters.size() > 0) {
+                    testName += String.format("(%s)", String.join(",", parameters));
```

Comment:
`"(" + String.join(",", parameters)) + ")" ` can be much faster.

Reply:
Make sense!